### PR TITLE
Update .json files of Deneyap Boards

### PR DESCRIPTION
**Pull request by DogushC**
Update .json files of Deneyap Boards

### CODE REVIEW

File: boards/deneyapkart.json [hunk 1]
```diff
@@ -7,19 +7,20 @@
...
       "-DBOARD_HAS_PSRAM",
+      "-DARDUINO_RUNNING_CORE=1",
+      "-DARDUINO_EVENT_RUNNING_CORE=1",
       "-mfix-esp32-psram-cache-issue",
       "-mfix-esp32-psram-cache-strategy=memw"
     ],
     "f_cpu": "240000000L",
     "f_flash": "80000000L",
-    "flash_mode": "qio",
+    "flash_mode": "dio",
```
Comment (valeros) [inline boards/deneyapkart.json:17]: Could you please explain the reasoning behind changing the `flash_mode` option? It seems `QIO` is used when selecting bootloader for the Deneyap boards in the Arduino core, is it wrong here https://github.com/espressif/arduino-esp32/blob/master/boards.txt#L17358 ?

File: boards/deneyapmini.json [hunk 2]
```diff
@@ -37,11 +35,9 @@
     "flash_size": "4MB",
     "maximum_ram_size": 327680,
     "maximum_size": 4194304,
-    "use_1200bps_touch": true,
-    "wait_for_upload_port": true,
```
Comment (valeros) [inline boards/deneyapmini.json:41]: https://github.com/espressif/arduino-esp32/blob/master/boards.txt#L17979
boards.txt states that this board requires 1200bps touch by default, is it wrong?

**Review by valeros — CHANGES_REQUESTED** [review]

**Comment by valeros** [comment]
Hi @DogushC , thanks for the PR. Please review comments above.

File: boards/deneyapkart.json (hunk 1 above)
Comment (DogushC) [reply boards/deneyapkart.json:17]: > Could you please explain the reasoning behind changing the `flash_mode` option? It seems `QIO` is used when selecting bootloader for the Deneyap boards in the Arduino core, is it wrong here https://github.com/espressif/arduino-esp32/blob/master/boards.txt#L17358 ?

I think that "flash_mode" parameter is equivalent at https://github.com/espressif/arduino-esp32/blob/master/boards.txt#L17357. 
If it is the equivalent of the "boot" parameter of arduino-esp32 repo, I have to update it here.

File: boards/deneyapmini.json (hunk 2 above)
Comment (DogushC) [reply boards/deneyapmini.json:41]: > https://github.com/espressif/arduino-esp32/blob/master/boards.txt#L17979 boards.txt states that this board requires 1200bps touch by default, is it wrong?

You are right. I will update it.

File: boards/deneyapkart.json (hunk 1 above)
Comment (valeros) [reply boards/deneyapkart.json:17]: > If it is the equivalent of the "boot" parameter of arduino-esp32 repo, I have to update it here.

Yes, it's equivalent, so I'd recommend using bootmode value from boards.txt in this field. The transformation to a proper `flash_mode` happens here https://github.com/platformio/platform-espressif32/blob/develop/builder/main.py#L68

File: boards/deneyapmini.json (hunk 2 above)
Comment (valeros) [reply boards/deneyapmini.json:41]: > You are right. I will update it.

The same with `deneyapminiv2`, seems like 1200bps is required.

File: boards/deneyapkart.json (hunk 1 above)
Comment (DogushC) [reply boards/deneyapkart.json]: Update all `flash_mode` parameter.

File: boards/deneyapmini.json (hunk 2 above)
Comment (DogushC) [reply boards/deneyapmini.json]: Fix `use_1200bps_touch` and `wait_for_upload_port` parameters of Deneyap Mini v2, Deneyap Mini and Deneyap Kart 1A v2.

Comment (DogushC) [reply boards/deneyapmini.json]: We have a [PR](https://github.com/espressif/arduino-esp32/pull/8046) for update `use_1200bps_touch` and `wait_for_upload_port` for Deneyap Kart 1A v2.

**Comment by valeros** [comment]
@DogushC do you plan any further updates?

**Comment by DogushC** [comment]
> @DogushC do you plan any further updates?

No. Thanks.

**Comment by valeros** [comment]
Thanks, merged.

**Pull request merged**